/faq/results broken links updates

ahodzic2 opened the pull request:
As part of PR https://github.com/corona-warn-app/cwa-website/pull/2656#issue-1189294825 I was able to run the check_anchor_links.js Cypress test for /faq/results/ and found a number of broken links.

Creating this PR to update these broken links to the corresponding new link.

Before:
![image](https://user-images.githubusercontent.com/72961430/161207745-a70e66d0-ffdc-4939-9f23-987959faaeb5.png)

After:
![image](https://user-images.githubusercontent.com/72961430/161207780-217d4897-8d98-449a-a07c-09a24919aece.png)

MikeMcC399 commented:
@ahodzic2 

Thanks for submitting this PR. It may result in some discussions, so it might have been better to open a separate issue first before opening this PR. I will take a look in any case.

MikeMcC399 commented:
- I have raised the question about the removed index anchors in #2661. It may be that all these references should be completely removed.

MikeMcC399 commented:
- This is the deduplicated list of errors prior to this PR, provided PR https://github.com/corona-warn-app/cwa-website/pull/2656 is used to run `check_anchor_links.js`:

```
http://localhost:8000/de/faq/results/#vac_cert_index on '/de/faq/results/' 
http://localhost:8000/de/faq/results/#test_cert_index on '/de/faq/results/' 
http://localhost:8000/de/faq/results/#vac_cert_booster on '/de/faq/results/' 
http://localhost:8000/de/faq/#howto on '/de/faq/results/' 
http://localhost:8000/de/faq/results/#check_in_index on '/de/faq/results/' 
http://localhost:8000/de/faq/#ene on '/de/faq/results/' 
http://localhost:8000/en/faq/results/#vac_cert_index on '/en/faq/results/' 
http://localhost:8000/en/faq/results/#test_cert_index on '/en/faq/results/' 
http://localhost:8000/en/faq/results/#vac_cert_booster on '/en/faq/results/' 
http://localhost:8000/en/faq/#android_location on '/en/faq/results/' 
http://localhost:8000/en/faq/#howto on '/en/faq/results/' 
http://localhost:8000/en/faq/results/#check_in_index on '/en/faq/results/' 
http://localhost:8000/en/faq/#ene on '/en/faq/results/' 
```

MikeMcC399 commented:
The link issues with these two should be resolved by changing the links to point to 
- https://www.coronawarn.app/de/#howto
- https://www.coronawarn.app/en/#howto

using relative links

```
http://localhost:8000/de/faq/#howto on '/de/faq/results/' 
http://localhost:8000/en/faq/#howto on '/en/faq/results/' 
```

MikeMcC399 commented on src/data/faq.json:
```diff
@@ -2410,7 +2406,7 @@
                                 "textblock": [
                                     "Yes. As long as you have activated exposure logging in the app and Bluetooth (and Location Services on Android) is on, encounters are logged in the background by the Exposure Notification system integrated in iOS/Android. The app itself doesn't have to be open for this. You can minimize it or use 'swipe-to-quit' to close it.",
-                                    "To be on the safe side, open the app once a day. You'll find more info at <a href='../#howto' target='_blank' >This is how the app works best</a>.",
+                                    "To be on the safe side, open the app once a day.",
```
resolve by changing relative link to point to top level of web, not by deleting the sentence.

MikeMcC399 commented on src/data/faq_de.json:
```diff
@@ -2419,7 +2415,7 @@
                                 "textblock": [
                                     "Ja. Solange Sie in der App die Risiko-Ermittlung aktiviert und Bluetooth (und bei Android die Standort-Dienste) eingeschalten haben, werden Begegnungen vom Exposure Notification Framework von iOS/Android aufgezeichnet. Die App selbst muss dafür nicht geöffnet bleiben. Sie können die App schließen, sowohl durch Minimieren als auch durch 'Swipe-to-quit'.",
-                                    "Sicherheitshalber können Sie die App einmal täglich öffnen. Weitere Hinweise finden Sie unter <a href='../#howto' target='_blank' >So funktioniert die App am besten</a>.",
+                                    "Sicherheitshalber können Sie die App einmal täglich öffnen.",
```
resolve by changing relative link to point to top level of web, not by deleting the sentence.

ahodzic2 replied:
Corrected as part of commit https://github.com/corona-warn-app/cwa-website/pull/2657/commits/62cef3b52d1590133787805b9c96a624138c2bcc

MikeMcC399 commented:
The resolution for these, changing to a pure anchor `#ene` looks good:

```
http://localhost:8000/de/faq/#ene on '/de/faq/results/' 
http://localhost:8000/en/faq/#ene on '/en/faq/results/' 
```

MikeMcC399 commented:
The correction for this one is also good:

`http://localhost:8000/en/faq/#android_location on '/en/faq/results/'`

MikeMcC399 reviewed (changes requested):
I suggest to modify the resolution for `#howto`.
A discussion is necessary to decide how to handle the index changes. See https://github.com/corona-warn-app/cwa-website/issues/2661.

ahodzic2 commented:
Hi @MikeMcC399 - Thank you for the thorough review and for the suggestion about opening an issue for discussion. I will look to address your suggestions later today after work and keep your suggestion in mind for future PRs/issues 😄 .

MikeMcC399 commented:
@ahodzic2 
>Thank you for the thorough review and for the suggestion about opening an issue for discussion. 

You're welcome. Probably I would have started with an issue, then submitted two different PRs. One for the links which had gone wrong due to relative references and a second one caused by the deleted lists of issues.

> I will look to address your suggestions later today after work and keep your suggestion in mind for future PRs/issues 😄 .

Sounds good.

One more point: I noticed that your commit message was quite verbose:

```
    /faq/results broken links updates

    As part of PR https://github.com/corona-warn-app/cwa-website/pull/2656#issue-1189294825 I was able to run the check_anchor_links.js Cypress test for /faq/results/ and found a number of broken links.

    Creating this PR to update these broken links to the corresponding new link.
```

If you compare to other commit messages, you will see that they are mostly one-line summaries. If you are making changes you might like to squash these and write a shorter commit message to go with it.

ahodzic2 commented:
@MikeMcC399  - I believe my latest commit addresses your comment about the `../../#howto` item. I thought the original item was `href=#howto` in reference to a removed FAQ section/anchor, so thank you for catching that.

I find myself once again having issues trying to squash but I do not want to simply close and open a new PR as there has been a lot of valuable conversation and references as part of this PR. There are some other PRs I noticed that had a few commits (outside of any merge commits) so I'm hoping an exception can be for this PR. For review purposes, one can change from 1 commit to all commits in the top left of the commit section.

I'll wait to hear about the index changes via the issue that you opened to see if an additional revision is needed.

MikeMcC399 commented:
@ahodzic2 
Thank you very much for all your effort on this set of issues so far!

- I have opened a new issue https://github.com/corona-warn-app/cwa-website/issues/2666 to suggest that all the links to "Questions and answers" articles are removed. For instance in https://www.coronawarn.app/en/faq/results/#vac_cert_multiple_scan_possible that would mean removing the link [To the index of the vaccination certificate](https://www.coronawarn.app/en/faq/results/#vac_cert_index) completely. I suggest to wait for feedback before making any more changes in this PR regarding those links.

MikeMcC399 commented:
@ahodzic2 
I will take a look at your problem with squashing. It is certainly possible to squash all commits when a PR is merged into the `master` branch, so if you can't squash the commits yourself, that does not invalidate the PR.

MikeMcC399 commented:
@ahodzic2 
One way of dealing with the challenge of squashing, when there are merges in the branch, is like this:
1. create a copy of your branch by creating a new branch based on `ahodzic2:faq_results_broken_links_update`
2. on your original local branch `ahodzic2:faq_results_broken_links_update` do a git hard reset to before your first commit
3. cherry pick your changes from the copied branch back to the original branch
4. squash your changes in the original branch
5. rebase your original branch on the `master `branch

(I did not get comfortable with doing this sort of thing until I did a more advanced training about `git`. I don't think I will ever get to the end of learning about `git`! It is bottomless!)

ahodzic2 commented:
> @ahodzic2 One way of dealing with the challenge of squashing, when there are merges in the branch, is like this:
> 
> 1. create a copy of your branch by creating a new branch based on `ahodzic2:faq_results_broken_links_update`
> 2. on your original local branch `ahodzic2:faq_results_broken_links_update` do a git hard reset to before your first commit
> 3. cherry pick your changes from the copied branch back to the original branch
> 4. squash your changes in the original branch
> 5. rebase your original branch on the `master `branch
> 
> (I did not get comfortable with doing this sort of thing until I did a more advanced training about `git`. I don't think I will ever get to the end of learning about `git`! It is bottomless!)

@MikeMcC399 

I will be saving these steps! I am definitely still a noob when it comes to git and I appreciate you passing this knowledge on to me. Will definitely keep this in mind and may try to play with this at some point following my busy weekend.



> @ahodzic2 Thank you very much for all your effort on this set of issues so far!
> 
> * I have opened a new issue [Superfluous links to deleted FAQ "Questions and answers" articles #2666](https://github.com/corona-warn-app/cwa-website/issues/2666) to suggest that all the links to "Questions and answers" articles are removed. For instance in https://www.coronawarn.app/en/faq/results/#vac_cert_multiple_scan_possible that would mean removing the link [To the index of the vaccination certificate](https://www.coronawarn.app/en/faq/results/#vac_cert_index) completely. I suggest to wait for feedback before making any more changes in this PR regarding those links.

Agreed! I will keep an eye on that issue to see what the desired route is by the community and open source team.

MikeMcC399 commented:
@larswmh 

I suggested in https://github.com/corona-warn-app/cwa-website/issues/2666 to delete multiple links instead of modifying them. Please take a look at this before anybody merges this PR.

dsarkar commented:
@MikeMcC399 https://github.com/corona-warn-app/cwa-website/pull/2657#issuecomment-1087458105 agreed, let's remove these links.

MikeMcC399 commented:
@ahodzic2 

- Could you please go ahead and implement the deletion of the links listed in https://github.com/corona-warn-app/cwa-website/issues/2666 as agreed with @dsarkar?
- Strangely a couple of the links `/en/faq/#ene` and `/en/faq/#android_location` are no longer causing issues. I would still leave the changes for these links in the PR though, as it is cleaner when changed.

MikeMcC399 reviewed (approved):
@larswmh It seems that you made the changes without waiting for @ahodzic2 

I ran the Cypress `check_anchor_links.js` test and there were no errors reported.

I am fine now with this PR and I think it can be merged.

dsarkar commented:
@MikeMcC399 Thanks for your review!
@ahodzic2 Thanks a lot for taking the initiative of this PR!
